client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx

```diff
@@ -23,6 +30,112 @@ import './style.scss';
 
 const debug = debugFactory( 'calypso:checkout-sidebar-plan-upsell' );
 
+const PromoCardV2 = styled.div`
+	position: relative;
+	background-color: #bbe0fa;
+	border-radius: 4px;
+	font-size: 12px;
+	font-weight: 500;
+	align-self: flex-end;
+
+	p {
+		margin: 0;
+		padding: 8px 16px;
+	}
+
+	&:before {
+		background-color: #bbe0fa;
+		display: block;
+		top: -5px;
+		content: '';
+		height: 8px;
+		right: 8%;
+		margin-left: -4px;
+		position: absolute;
+		transform: rotate( 225deg );
+		width: 8px;
+	}
+`;
+
+const CheckoutPromoCard: React.FC< {
+	onUpgradeClick: () => void;
+	responseCart: ResponseCart;
+	currentVariant: WPCOMProductVariant;
+	percentSavings: number;
+} > = ( { onUpgradeClick, responseCart, currentVariant, percentSavings } ) => {
+	const translate = useTranslate();
+	const plan = responseCart.products.find(
+		( product ) => isPlan( product ) && ! isJetpackPlan( product )
+	);
+
+	const domainRegistrationOrTransferInCart = responseCart.products.find(
+		( product ) => isDomainRegistration( product ) || isDomainTransfer( product )
+	);
+
+	const isMonthly = currentVariant?.termIntervalInMonths === 1;
+	const isYearly = currentVariant?.termIntervalInMonths === 12;
+	const isBiennially = currentVariant?.termIntervalInMonths === 24;
+	const isTriennially = currentVariant?.termIntervalInMonths === 36;
+
+	let labelText;
+
+	// TODO: this currently doesn't work because the !plan guard found in CheckoutSidebarPlanUpsell disallows this to be reached.
```

Comment:
This one is a bit tricky to figure out because I haven't found a way to get to this point without a plan in the cart, since code in CheckoutSidebarPlanUpsell relies on a plan existing in the cart. WIP

Reply:
I'm trying to grok why this is here. Before this PR, the logic was to show this notice if `plan && shouldUseCheckoutV2`, but now we show it if `! plan && domainRegistrationOrTransferInCart`? I know that this component isn't rendered unless `shouldUseCheckoutV2` is true, but why invert the `plan` condition and add `domainRegistrationOrTransferInCart`?

Reply:
Ah, so this is an attempt to refactor the 'free plan upsell' and move it from SecondaryCartPromotions into this section, since its effectively just a billing term upsell/upgrade.

I think shimming it into this new component would require a larger rewrite since all of the current conditions require a plan and this one requires the absence of a plan. Unfortunately, trying to support v1 `PromoCard`s functionality requires us to do the opposite of what I am trying to do here and it creates a logic nightmare. I may hold off on moving the free plan upsell until after v2 is merged and theres less competing logic.